File: frontend/src/types/click.ts
export interface Click {
  id: number;
  clicked_at: string;
  country: string | null;
  city: string | null;
  device_type: string | null;
  browser: string | null;
  os: string | null;
}

export interface LinkClickEvent extends Click {
  click_count: number;
}

File: backend/app/schemas/click.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ClickRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    clicked_at: datetime
    country: str | None
    city: str | None
    device_type: str | None
    browser: str | None
    os: str | None
